File: remote/src/main/java/com/kelsos/mbrc/repository/TrackRepository.java
package com.kelsos.mbrc.repository;

import android.graphics.Bitmap;
import com.kelsos.mbrc.domain.TrackPosition;
import com.kelsos.mbrc.dto.track.Rating;
import com.kelsos.mbrc.dto.track.TrackInfo;
import java.util.List;
import rx.Observable;

public interface TrackRepository {
  Observable<TrackInfo> getTrackInfo(boolean reload);

  Observable<List<String>> getTrackLyrics(boolean reload);

  Observable<Bitmap> getTrackCover(boolean reload);

  Observable<TrackPosition> getPosition();

  void setPosition(TrackPosition position);

  Observable<Rating> getRating();

  void setRating(Rating rating);

  void setTrackInfo(TrackInfo trackInfo);

  void setLyrics(List<String> lyrics);

  void setTrackCover(Bitmap cover);
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/repository/TrackRepository.kt
package com.kelsos.mbrc.repository

import android.graphics.Bitmap
import com.kelsos.mbrc.domain.TrackInfo
import com.kelsos.mbrc.domain.TrackPosition
import com.kelsos.mbrc.domain.TrackRating
import rx.Observable

interface TrackRepository {
    fun getTrackInfo(reload: Boolean): Observable<TrackInfo>

    fun getTrackLyrics(reload: Boolean): Observable<List<String>>

    fun getTrackCover(reload: Boolean): Observable<Bitmap?>

    fun getPosition(): Observable<TrackPosition>

    fun setPosition(position: TrackPosition)

    fun getRating(): Observable<TrackRating>

    fun setRating(rating: TrackRating)


    fun setTrackInfo(trackInfo: TrackInfo)

    fun setLyrics(lyrics: List<String>)

    fun setTrackCover(cover: Bitmap)
}
